components/gantry/singleaxis/singleaxis.go

```diff
@@ -631,9 +631,15 @@ func (g *singleAxis) CurrentInputs(ctx context.Context) ([]referenceframe.Input,
 }
 
 // GoToInputs moves the gantry to a goal position in the Gantry frame.
-func (g *singleAxis) GoToInputs(ctx context.Context, goal []referenceframe.Input) error {
-	g.mu.Lock()
-	defer g.mu.Unlock()
-	speed := []float64{}
-	return g.MoveToPosition(ctx, referenceframe.InputsToFloats(goal), speed, nil)
+func (g *singleAxis) GoToInputs(ctx context.Context, inputSteps ...[]referenceframe.Input) error {
+	for _, goal := range inputSteps {
+		g.mu.Lock()
+		defer g.mu.Unlock()
```

Comment:
same defer question

Reply:
Ack, this would definitely deadlock.

Reply:
why didn't tests catch it? Could we please add tests?

Reply:
:+1: